### PR TITLE
More like yapapi

### PR DESCRIPTION
**Pull request by filipgolem**

### CODE REVIEW

File: yajsapi/runner/index.ts [hunk 1]
```diff
@@ -695,9 +697,17 @@ export class Engine {
...
+        let done_workers = new Set([...workers].filter((worker) => !worker.isPending()));
+
         workers = new Set([...workers].filter((worker) => worker.isPending()));
         services = services.filter((service) => service.isPending());
 
+        for (let worker_task of [...done_workers]) {
+            if (worker_task.isPending()) {
+                await worker_task;
+            }
+        }
```
Comment (azawlocki) [inline yajsapi/runner/index.ts:709]: This does nothing: `done_workers` is the set of workers for which `worker.isPending()` is false, so the condition in `if` is never satisfied (assumming a worker cannot change from `! isPending()` to `isPending()`).

File: yajsapi/runner/index.ts [hunk 2]
```diff
@@ -719,21 +728,27 @@ export class Engine {
           for (let worker_task of [...workers]) {
             worker_task.cancel();
           }
-          // await asyncio.wait(workers, timeout=15, return_when=asyncio.ALL_COMPLETED)
+          await bluebird.Promise.any([
+            bluebird.Promise.all([...workers]),
+            promise_timeout(10)
+          ]);
```
Comment (azawlocki) [inline yajsapi/runner/index.ts:734]: I don't know `bluebird.Promise`, but this looks suspicious: shouldn't it be `bluebird.Promise.all(...)`?

File: yajsapi/runner/index.ts [hunk 3]
```diff
@@ -719,21 +728,27 @@ export class Engine {
...
       await bluebird.Promise.any([
-        Promise.all([process_invoices_job]),
-        promise_timeout(15),
+        bluebird.Promise.all([
+          find_offers_task,
+          process_invoices_job,
+        ]),
+        promise_timeout(10)
       ]);
+      if (agreements_to_pay.size == 0) {
```
Comment (azawlocki) [inline yajsapi/runner/index.ts:746]: In `yapapi` we wait for `process_invoices_job` only if `agreements_to_pay` is **not empty**.

**Review by azawlocki — CHANGES_REQUESTED** [review]

File: yajsapi/runner/index.ts (hunk 3 above)
Comment (filipgolem) [reply yajsapi/runner/index.ts:746]: Good catch!

Comment (filipgolem) [reply yajsapi/runner/index.ts]: Changed to > 0

File: yajsapi/runner/index.ts (hunk 1 above)
Comment (filipgolem) [reply yajsapi/runner/index.ts:709]: Removed.

**Review by azawlocki — APPROVED** [review]

**Pull request merged**